File: du-backend/src/main/java/com/dwin/du/api/entity/Product.java
package com.dwin.du.api.entity;
import jakarta.persistence.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "receipt_id")
    private Receipt receipt;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "price", nullable = false)
    private Double price;

    @Column(name = "divisible")
    private boolean divisible;

    @Column(name = "max_quantity")
    private int maxQuantity;

    @Column(name = "compensation_price")
    private double compensationPrice;

    @Column(name = "is_settled")
    private boolean settled;
}

File: divvyup/DivvyUp.Shared/Model/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DivvyUp_Shared.Model
{
    public class Product
    {
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("Receipt")]
        [Column("receipt_id")]
        public int ReceiptId { get; set; }
        public Receipt Receipt { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; }

        [Required]
        [Column("price")]
        public decimal Price { get; set; }

        [Required] 
        [Column("is_divisible")] 
        public bool Divisible { get; set; }

        [Required]
        [Column("max_quantity")]
        public int MaxQuantity { get; set; }

        [Column("compensation_price")]
        public decimal CompensationPrice { get; set; }

        [Column("is_settled")]
        public bool Settled { get; set; }
    }
}
